### PR TITLE
feat: use map for azureIdentities instead of list in helm chart

### LINKED ISSUE #885: Helm chart should consider using map rather than array for "azureIdentities"
**Describe the request**
aad-pod-identity Helm chart currently supports something like this in `values.yaml`:
```
  azureIdentities:
    - name: aso-identity
      type: 0
      # Update values for Managed Identity
      resourceID: ""
      clientID: ""
      binding:
        name: aso-identity-binding
        selector: aso_manager_binding
```

The problem is that Helm doesn't do a great job managing arrays, as you can see discussed in this issue: https://github.com/helm/helm/issues/1987, for example it seems that when using `--set azureIdentities[0].xyz` you cannot merge the contents of `values.yaml` and `--set`. If you want to say set just a few fields of an identity (maybe the `resoruceID` and `clientID`), you actually end up having to set _all_ the properties since Helm doesn't do a nice merge.

@technosophos suggested in that same thread that you could prefer a map in these instances. That in theory should allow `--set` and `values.yaml` contents to be merged like it normally is.

**Explain why AAD Pod Identity needs it**
Would make it significantly easier to manage aad-pod-identity as a subchart, or to make use of standard `--set` and `values.yaml` merging provided by Helm.

Do note that this would be a breaking chart change I think.

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

fixes #885 

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Notes for Reviewers**:

### CODE REVIEW

**Review by chewong — CHANGES_REQUESTED** [review]

File: manifest_staging/charts/aad-pod-identity/values.yaml [hunk 1]
```diff
@@ -211,19 +211,30 @@ rbac:
...
-azureIdentities: []
-  # - name: "azure-identity"
+# This is a map with the AzureIdentityName being the key and the rest of the blob as value in accordance
+# to helm best practices: https://helm.sh/docs/chart_best_practices/values/#consider-how-users-will-use-your-values
+azureIdentities:
+  # "azure-identity":
   #   # if not defined, then the azure identity will be deployed in the same namespace as the chart
   #   namespace: ""
-  #   # type 0: MSI, type 1: Service Principal
+  #   # type 0: MSI, type 1: Service Principal, type 2: Service principal with certifcate
```
Comment (chewong) [inline manifest_staging/charts/aad-pod-identity/values.yaml:220]:
```suggestion
  #   # type 0: User-assigned identity, type 1: Service Principal, type 2: Service principal with certificate
```

File: manifest_staging/charts/aad-pod-identity/templates/identities.yaml [hunk 2]
```diff
@@ -2,27 +2,36 @@
...
+      clientID: {{ required "clientID is required!" $value.clientID }}
+      {{- if eq (int $value.type) 0 }}
+      resourceID: {{ required "resourceID is required!" $value.resourceID }}
+      {{- else }}
+      tenantID: {{ required "tenantID is required!" $value.tenantID }}
+      clientPassword: {{ required "clientPassword is required!" $value.clientPassword }}
+      auxiliaryTenantIDs:
+      {{- range $value.auxiliaryTenantIDs }}
+        - {{ . }}
+      {{- end }}
```
Comment (chewong) [inline manifest_staging/charts/aad-pod-identity/templates/identities.yaml:24]: for readability:

```suggestion
        {{- range $value.auxiliaryTenantIDs }}
        - {{ . }}
        {{- end }}
```

File: manifest_staging/charts/aad-pod-identity/values.yaml [hunk 3]
```diff
@@ -211,19 +211,30 @@ rbac:
...
+  #   # Required for type 1 and 2
+  #   clientPassword: {}
+  #   # Optional for type 1 and 2 (multi-tenant)
+  #   auxiliaryTenantIDs: []
   #   binding:
   #     name: "azure-identity-binding"
   #     # The selector will also need to be included in labels for app deployment
   #     selector: "demo"
 
+
```
Comment (chewong) [inline manifest_staging/charts/aad-pod-identity/values.yaml:238]: nit: remove extra line

File: manifest_staging/charts/aad-pod-identity/values.yaml [hunk 4]
```diff
@@ -211,19 +211,30 @@ rbac:
...
   #   type: 0
   #   # /subscriptions/subscription-id/resourcegroups/resource-group/providers/Microsoft.ManagedIdentity/userAssignedIdentities/identity-name
+  #   # Required for type 0
   #   resourceID: ""
+  #   # Required for type 0, 1 and 2
   #   clientID: ""
+  #   # Required for type 1 and 2
+  #   tenantID: ""
+  #   # Required for type 1 and 2
+  #   clientPassword: {}
```
Comment (chewong) [inline manifest_staging/charts/aad-pod-identity/values.yaml:230]:
```suggestion
  #   clientPassword:
  #     name: ""
  #     namespace: "" 
```

**Review by chewong — APPROVED** [review]
LGTM with a few nits

File: manifest_staging/charts/aad-pod-identity/values.yaml (hunk 4 above)
Comment (aramase) [reply manifest_staging/charts/aad-pod-identity/values.yaml]: `clientPassword` is not a map. It is [types.NamespacedName](https://pkg.go.dev/k8s.io/apimachinery/pkg/types#NamespacedName). I've updated the PR with the format.

**Pull request merged**